vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SlideManager.java

```diff
@@ -167,4 +186,74 @@ public Page<ISlide> findByNameOrDescription(Pageable requestedPage, String searc
 
         return slideRepo.findDistinctByNameContainingOrDescriptionContaining(requestedPage, searchText,searchText);
     }
+
+    @Override
+    public void updateNameAndDescription(ISlide slide, SlideForm slideForm) {
+        slide.setName(slideForm.getDefaultName().getText());
+        slide.setDescription(slideForm.getDefaultDescription().getText());
+        addSlideName(slide,slideForm.getDefaultName());
+        addSlideDescription(slide,slideForm.getDefaultDescription());
+        
+        for(LocalizedTextForm title:slideForm.getNames()) {        
+            addSlideName(slide,title);
+        }
+        for(LocalizedTextForm text:slideForm.getDescriptions()) {
+            addSlideDescription(slide,text);
+        }
+    }
+    
+    /**
+     * Adds name to slideNames List
+     * @param slide
+     * @param names
+     */
+    @Override
+    public void addSlideName(ISlide slide, LocalizedTextForm name) {
+        if(name!=null && !StringUtils.isEmpty(name.getText())) {
```

Comment:
same as above, not a requirement

Reply:
The  method "updateNameAndDescription" in SlideManager uses addSlideName.

Reply:
but why can name not be empty?